sourcefiles/randoconfig.py

```diff
@@ -1183,64 +1208,71 @@ def __init__(self, rom: bytearray = None):
             self.enemy_atkdb = enemytechdb.EnemyAttackDB.from_rom(rom)
             self.enemy_aidb = enemyai.EnemyAIDB.from_rom(rom)
 
-    def to_json(self, outfile_name):
-        # Make key item dict
-        # self.key_item_locations uses logictypes.Location (and subclasses)
-        # instead of tids because of linked locations
-        key_item_dict = {
-            loc.getName(): str(loc.getKeyItem())
-            for loc in self.key_item_locations
-        }
+    def _jot_json(self):
+        def enum_key_dict(d):
+            "Properly uses str(key) for dicts with StrIntEnum keys."
+            return { str(k): v for (k,v) in d.items() }
```

Comment:
This stuff is annoying. Since our enums are based on IntEnum, they're technically ints, which means we can't customize the JSONEncoder behavior for them. thus, calling `str()` a bunch. Might be a nicer way to do it, but anyway.